test/Modules/MatrixGroupExperimental-test.jl

```diff
@@ -0,0 +1,10 @@
+@testset "Modules" begin
+  _wrap_for_gap(m::MatrixElem) = GAP.Globals.MakeJuliaMatrixRep(m)
+
+    m = matrix(ZZ, [0 1 ; -1 0])
+    gapM = _wrap_for_gap(m)
+    G = GAP.Globals.Group(gapM)
+    @test GAP.Globals.Size(G) == 4
```

Comment:
This enumerates the group. Unfortunately this wants to compare matrices with `<` which is not implemented for most (all) Oscar matrices. I'll write more about this elsewhere; but for now, I'd rather try to *avoid* this, by giving GAP a better way to compute the matrix:

Let's create and set a `NiceMonomorphism` for the group `G`!

Reply:
See also https://github.com/oscar-system/Oscar.jl/pull/538#discussion_r673856914
```
gap> SetNiceMonomorphism(G, hom);
gap> SetIsHandledByNiceMonomorphism(G, true);
gap> Size(G);
4
```
How to create `hom`? I would use `GroupHomomorphismByFunction` and then use the existing conversion functions for matrices we provide in Oscar, see the files in `src/GAP/`. Talk to me if you need help finding stuff, it is unfortunately not documented perfectly *ahem*

Reply:
For number fields, of course we need something more sophisticated, but even there we already have code. But let's start with support for ZZ, QQ, and finite *prime* fields

Reply:
Further relevant notes:
- relevant files are `src/GAP/iso_gap_oscar.jl` and `src/GAP/iso_oscar_gap.jl`
- these contain functions to create morphisms between GAP and OSCAR rings
- in our examples we start with Oscar matrices, so `_iso_oscar_gap` is the function to use -- apply it to the `base_ring` of the matrices to get a map
- this map can be applied to Oscar matrices `AbstractAlgebra.map_entries`, for the reverse direction use `preimage_matrix`
- use this to create a `GroupHomomorphismByFunction` (make sure to use the variant with four arguments: domain, codomain/range, function to compute images, function to compute preimages)

Reply:
Further relevant notes:
- relevant files are `src/GAP/iso_gap_oscar.jl` and `src/GAP/iso_oscar_gap.jl`
- these contain functions to create morphisms between GAP and OSCAR rings
- in our examples we start with Oscar matrices, so `_iso_oscar_gap` is the function to use -- apply it to the `base_ring` of the matrices to get a map
- this map can be applied to Oscar matrices `AbstractAlgebra.map_entries`, for the reverse direction use `preimage_matrix`
- use this to create a `GroupHomomorphismByFunction` (make sure to use the variant with four arguments: domain, codomain/range, function to compute images, function to compute preimages)